### DISCUSSION TITLE
tuya cloud device id and key and OTA

### DISCUSSION THREAD

**Discussion by vandinter** [ct-Open-Source/tuya-convert#1032] [Ideas]
Not sure if this was looked into already and it sounds too obvious for me - but you never know, so I give it a try: When connecting a tuya device to the tuya cloud and setting up a free developer account, connecting this account to the tuya app used to set up the device one can read the individual device id and key from the tuya developer portal... won'T that help with re-enabeling OTA?

**Comment by vandinter** [comment]
@kueblc Any thoughts on that?

**Reply by techman83** [reply]
The [wiki](https://github.com/ct-Open-Source/tuya-convert/wiki/Collaboration-document-for-PSK-Identity-02) contains everything that has been tried so far.

**Reply by vandinter** [reply]
Hi @techman83 , thanks I read that and therefore was wondering if one could not use the tuya developer cloud and extracting the device information from the cloud would allow to get the necessary connection secrets. TuyaApi is using this method after all to communicate with tuya stock firmware on a local network...
